src/vivarium/framework/engine.py

```diff
@@ -322,6 +327,9 @@ def add_components(self, component_list: List[Component]) -> None:
     def get_population(self, untracked: bool = True) -> pd.DataFrame:
         return self._population.get_population(untracked)
 
+    def past_stop_time(self) -> bool:
```

Comment:
nit: why the indirection?

Also, maybe flipping the logic back and naming it `before_stop_time` is clearer? (1) you don't need the not and (2) technically you're not "past" the stop time if the clock time == stop time.

Reply:
We use this again in `vivarium_cluster_tools`, because I wanted to add the new `run` method incorporating the backups there instead of from simulationcontext. So this is just to avoid using private attributes from outside SimulationContext.

Reply:
originally, i thought putting the e.g. `run_with_backups` method in vivarium wouldn't be feasible, but it definitely seems possible. maybe check the PR there and see if you'd rather we incorporate that function as a method here and simply call it from VCT

Reply:
removed